### DISCUSSION TITLE
Feedback from a fellow paperless-ngx user

### DISCUSSION THREAD

**Discussion by jonaswinkler** [astubenbord/paperless-mobile#104] [Feature Requests]
Hello!

This is a great application, and its easy to notice you've put a lot of effort and thought into it! It's great so see things like this happen :)

I do have a couple recommendations through on what to improve, in no particular order, and looking at the app as a naive user. Maybe you find some of them useful! Most of them are just minor nitpicks though.

* When searching for document, debounce input in the search bar. This means that a server request to load autocomplete recommendations is only made when the user stopped typing (for about a second, maybe). I've had cases where I was typing "amazo", and the first recommendation is "amazing", since loading recommendations for "amaz" took longer than loading recommendations for "amazo".

* Autocompleting multiple words would be nice. Google has a very elegant solutions on their search boxes, touching a recommendation searches for it, touching the arrow next to it copies it (with a trailing space) to the search bar to continue typing in the query.

* ASN: I'd prefer an option to manually enter (and edit) the ASN of any document. I don't really know how users use this, but the way I initially intented this to work was in the following way:

  - I have a binder for documents which I need to keep physically, all of them have an ASN, and are sorted by that.
  - I write ASNs on important documents before scanning them, and then add them to the binder.
  - When I get to sorting the inbox, which might be days or weeks later, I assign the ASN found on the scan of the document to the doucment metadata.

Maybe you can enlighten me how you use it, and I can adjust my workflow.

* I think search results and similar documents are sorted by date? I'd consider sorting them by relevance. This way, documents that better match the search query (or are the most similar to a document) appear first. Just pass ordering=score to the api.

* Also related to the search results, the API also sends fragments of the document content with the search terms highlighted, formatted in HTML. Its visible in the web app on search results when you select the large cards. Combine that with sorting by relevance for even better search :)

* On the document details page, tapping the partial preview image on the top above the tabs could open the preview.

* Sometimes, the document edit button has a badge "1" attached to it. I figured this appears when there is an edit recommendation? I like the idea, however, this also appears on documents where I verified that the date / correspondent / etc is correct. Sometimes, the date parsing logic finds multiple dates in a document, and those are getting recommended, resulting in the badge, and implying that I still need to edit the document. As a workaround, maybe only display the badge when there's an inbox tag on the document? Not sure.

* In the document list, when selecting the most compact "list" view type, there's lots of border around the small preview images. I think it wouldn't hurt to make the borders smaller, resulting in larger previews.

And lastly, some weird observation I made, where I'm not sure if its a bug or not.

* My paperless instance is on my home network. When I'm using your app while being connected to the home network via a wireguard VPN tunnel (this is how I access my services anywhere), some parts of the application are greyed out / missing. This includes the document edit button (missing), the entries in the saved views list (greyed out) and ability to upload files (greyed out). I haven't quite figured out yet what's going on here. Maybe some sort of network check that fails?

Best regards,
Jonas

**Comment by astubenbord** [comment]
Hi Jonas, 

first of all thanks for your kind words, I really do appreciate it! It's always nice to get some valuable feedback from users, and active contributors of Paperless (you especially) that know the capabilities of paperless-ng(x) and have it integrated into their lives. I currently do not yet have the need to archive and index that many documents, therefore I would consider myself a very basic/naive user who doesn't use most of Paperless's capabilities (as of now, I'm sure that will change in the near future, thanks to good old German bureaucracy).

> When searching for document, debounce input in the search bar...

> Autocompleting multiple words would be nice...

These are very good ideas and I will consider them in the next release cycles. The search is fairly new and I basically just implemented the M3 design spec.

> ASN: I'd prefer an option to manually enter (and edit) the ASN of any document. I don't really know how users use this, but the way I initially intented this to work was in the following way:

That's good to know! I am not really familiar with the workflows of using and assigning ASNs, I've followed some discussions here and there and there seems to be multiple ways of doing so. I agree that the way it is currently implemented is highly opinionated and doesn't allow for much flexibility, which probably had to do with a lack of understanding. I think it might be a good idea to also display the next available ASN on the scanner page before any pages have been scanned. What do you think about that? Would that be a benefit or do you essentially only look at the last physical ASN on the latest document in your binder?

Just out of curiosity, if this is the intended workflow, wouldn't it be beneficial to also be able to specify the ASN on document upload (just like correspondent, created date, tags, and document type)? Or are there technical limitations to this approach?

> I think search results and similar documents are sorted by date? I'd consider sorting them by relevance.

You're right, it is currently sorted by date, I will adjust this.

> On the document details page, tapping the partial preview image on the top above the tabs could open the preview.

This should also be very easy to implement and fairly obvious to the user once implemented. I've had requests that scrolling down would open the preview, but I think just tapping is easier, maybe not as fancy but functional.

> Also related to the search results, the API also sends fragments of the document content with the search terms highlighted, formatted in HTML. Its visible in the web app on search results when you select the large cards. Combine that with sorting by relevance for even better search :)

I've seen this and part of it is already implemented. However, from what I've seen the response is html format, I'll have to see how that complements with the rest of the app as I've not really played around with the HTML renderer. But, specifically with the "new" detailed view, this might be a useful addition to make it actually detailed :D

> My paperless instance is on my home network. When I'm using your app while being connected to the home network via a wireguard VPN tunnel (this is how I access my services anywhere), some parts of the application are greyed out / missing. This includes the document edit button (missing), the entries in the saved views list (greyed out) and ability to upload files (greyed out). I haven't quite figured out yet what's going on here. Maybe some sort of network check that fails?

Hm, I've never used it with a running VPN, but greyed out buttons mean that your device doesn't have a connection to the internet. Could have to do with the VPN, but I'm not sure.


Best regards, 
Anton

**Reply by jonaswinkler** [reply]
> I am not really familiar with the workflows of using and assigning ASNs, I've followed some discussions here and there and there seems to be multiple ways of doing so.

You can have a look at the documentation, it has some tips regarding this. https://docs.paperless-ngx.com/usage/#usage-recommended-workflow

> I think it might be a good idea to also display the next available ASN on the scanner page before any pages have been scanned.

This is a good idea. However, when you're scanning that physical binder is probably not that far away, and I'd say the impact of this feature won't be that big.

> Just out of curiosity, if this is the intended workflow, wouldn't it be beneficial to also be able to specify the ASN on document upload.

Yes! Not sure if the API can do that right now.

**Reply by astubenbord** [reply]
> Yes! Not sure if the API can do that right now.

Thanks for the feedback, I'll open a feature request in paperless-ngx and see what the community / other maintainers have to say :)

**Comment by astubenbord** [comment]
Ah, regarding this:

>Sometimes, the document edit button has a badge "1" attached to it. I figured this appears when there is an edit recommendation? I like the idea, however, this also appears on documents where I verified that the date / correspondent / etc is correct. Sometimes, the date parsing logic finds multiple dates in a document, and those are getting recommended, resulting in the badge, and implying that I still need to edit the document. As a workaround, maybe only display the badge when there's an inbox tag on the document? Not sure.

I'm not entirely sure what's the best option here. Maybe only show a badge if there exist suggestions for an unset field, i.e. if only correspondent is not set and there are suggestions for correspondents, show a "1" and so on... 

At least from my intention, inbox tags should be handled in the dedicated inbox.
